aten/src/ATen/native/DilatedMaxPool2d.cpp

```diff
@@ -154,8 +153,9 @@ const Tensor& indices) {
     outputHeight_for_shape_check, outputWidth_for_shape_check,
     memory_format);
 
-  set_output_raw_strided(0, input.sizes(), {}, input.options().memory_format(memory_format),
-             input.has_names() ? input.names() : DimnameList{});
+  // FIXME: the last argument has to be the input names
+  set_output_raw_strided(0, input_sizes, {}, gradOutput.options().memory_format(memory_format),
+             DimnameList{});
```

Comment:
Can't do that, with the above there was a test failing even for `pytest test/test_ops_gradients.py -vk max_pool2d`.

Reply:
if there's a test that test the names version of this function, I think it'd be fine to remove it really. There's been some PRs trying to remove named tensors completely from the codebase (e.g. https://github.com/pytorch/pytorch/pull/82158) so removing those tests is fine really.

Reply:
@lezcano Sorry, I was not clear there: I meant to say that passing input memory format as an argument to the backward function is necessary.  If `gradOutput.options().memory_format(memory_format)` used instead, there was a test failing with `pytest test/test_ops_gradients.py -vk max_pool2d`.  I haven't encountered any test failing because of not assigning names to the gradient output, so it should be safe just to not assign any names to the resulting gradient tensor.